### ISSUE TITLE
pulumi refresh causes auth0.ResourceServerScopes resource to undeploy from Resource Server on next plan

### ISSUE THREAD

**Issue by 1tmb** [pulumi/pulumi-auth0#309]
### What happened?

After I run `pulumi refresh`, `pulumi preview --diff` wishes to remove ` auth0.ResourceServerScopes` from my resource server despite no changes to the resource service, the scopes object or indeed anything else.

`pulumi up` then removes the resource.

Subsequent invocations of `pulumi preview` then fail to notice that I still have the `auth0.ResourceServerScopes` in my markup, so it should be looking to deploy that. Yet the preview ignores it entirely as the state file is out of sync.

The only way I can get it to deploy is to make a change to one of the roles to trick it into planning and deploying, and then change it back to deploy the actual configuration I want.

This is however short lived because at random a subsequent plan will just undeploy it anyway.

### Example

const scopes = new auth0.ResourceServerScopes("backendApiScopes", {
        resourceServerIdentifier: backendApiResourceId,
        scopes: [
            {
                description: "Create a user within a tenancy",
                name: "create:user",
            }
        ],
    })

### Output of `pulumi about`

```
CLI          
Version      3.86.0
Go Version   go1.21.1
Go Compiler  gc

Plugins
NAME    VERSION
auth0   2.24.3
nodejs  unknown

Host     
OS       fedora
Version  38
Arch     x86_64
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
Hi @1tmb this sounds similar to this upstream issue: https://github.com/auth0/terraform-provider-auth0/issues/687

It looks like a fix is available in our upcoming 3.0 version of the provider, which we are working on today. Please give that one a try when it's out.

**Comment by 1tmb** [comment]
Working through v3.0.0 breaking changes to retest, will report back.
